sql/src/test/java/org/apache/druid/sql/calcite/run/SqlResultsTest.java

```diff
@@ -222,6 +224,17 @@ public void testCoerceString()
     assertCannotCoerce(new Object(), SqlTypeName.VARCHAR);
   }
 
+  @Test
+  public void testCoerceOfArrayOfPrimitives()
+  {
+    try {
+      assertCoerce("", new byte[1], SqlTypeName.BIGINT);
+      Assert.fail("Should throw an exception");
+    }
+    catch (Exception e) {
+      Assert.assertEquals("Cannot coerce field [fieldName] class of [[B] to [BIGINT]", e.getMessage());
```

Comment:
weren't we going to replace `[[B]` with a more friendly name such as binary?

Reply:
added mapping